### PR TITLE
[Vulkan] Improved SPIRV-Tools library search on Linux

### PR DESCRIPTION
**Pull request by masahi**
Related issue = #2298

I got a undefined symbol error on Linux/Vulkan, this change made things working for me and I got taichi demo working on my laptop AMD APU with vk backend!

cc @bobcao3 @k-ye

### CODE REVIEW

File: cmake/TaichiCore.cmake [hunk 1]
```diff
@@ -223,10 +223,16 @@ if (TI_WITH_VULKAN)
       find_library(SPIRV_OPT NAMES "SPIRV-Tools-opt" PATHS "${Vulkan_INCLUDE_DIR}/../Lib" REQUIRED)
       find_library(SPIRV_LINK NAMES "SPIRV-Tools-link" PATHS "${Vulkan_INCLUDE_DIR}/../Lib" REQUIRED)
     else ()
-      set(SPIRV_OPT SPIRV-Tools-opt)
-      set(SPIRV_TOOLS SPIRV-Tools)
-      set(SPIRV_LINK SPIRV-Tools-link)
+      get_filename_component(VULKAN_LIBRARY_PATH ${Vulkan_LIBRARY} DIRECTORY)
+      find_library(SPIRV_TOOLS NAMES "SPIRV-Tools"  PATHS ${VULKAN_LIBRARY_PATH} REQUIRED)
+      find_library(SPIRV_OPT NAMES "SPIRV-Tools-opt" PATHS ${VULKAN_LIBRARY_PATH} REQUIRED)
+      find_library(SPIRV_LINK NAMES "SPIRV-Tools-link"  PATHS ${VULKAN_LIBRARY_PATH} REQUIRED)
```
Comment (bobcao3) [inline cmake/TaichiCore.cmake:229]: Is this an variable from the environment? Or is it a manually set flag

Comment (masahi) [reply cmake/TaichiCore.cmake:229]: See L226. The directory path is extracted from ${Vulkan_LIBRARY}.

Comment (bobcao3) [reply cmake/TaichiCore.cmake:229]: Okay sorry missed that

**Comment by bobcao3** [comment]
Looks good

**Review by bobcao3 — APPROVED** [review]

**Pull request merged**